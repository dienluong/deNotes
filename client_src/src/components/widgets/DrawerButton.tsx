import React from 'react';
import Fab from '@material-ui/core/Fab';
import MenuIcon from '@material-ui/icons/Menu';
import Zoom from '@material-ui/core/Zoom';

function DrawerButton({ visible, clickHandler, className }: { visible: boolean, clickHandler: () => unknown, className: string }) {
  return (
    <div className={ className }>
      <Zoom in={ visible } unmountOnExit>
        <Fab aria-label="notes list drawer" onClick={ clickHandler }>
          <MenuIcon />
        </Fab>
      </Zoom>
    </div>
  );
}

export default DrawerButton;
