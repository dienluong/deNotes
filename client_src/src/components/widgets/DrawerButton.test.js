import React from 'react';
import DrawerButton from './DrawerButton';
import Fab from '@material-ui/core/Fab';
import MenuIcon from '@material-ui/icons/Menu';
import Zoom from '@material-ui/core/Zoom';

// Configure Enzyme
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
let wrapper = {};

afterEach(() => {
  if (typeof wrapper.unmount === 'function') {
    wrapper.unmount();
  }
});

it('should render button with material-ui Zoom and Fab components', () => {
  const clickHandler = jest.fn();
  const props = {
    size: 'small',
    visible: true,
    clickHandler,
    className: 'some-css-class',
  };

  wrapper = shallow(<DrawerButton { ...props } />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Zoom).exists()).toBe(true);
  expect(wrapper.find(Zoom).prop('in')).toBe(props.visible);
  expect(wrapper.find(Fab).exists()).toBe(true);
  expect(wrapper.find(Fab).prop('size')).toBe(props.size);
  expect(wrapper.find(MenuIcon).exists()).toBe(true);
});

it('should react to click by invoking handler function', () => {
  const clickHandler = jest.fn();
  const props = {
    size: 'small',
    visible: true,
    clickHandler,
    className: 'some-css-class',
  };

  wrapper = shallow(<DrawerButton { ...props } />);
  wrapper.find(Fab).simulate('click');
  expect(clickHandler).toBeCalledTimes(1);
});
