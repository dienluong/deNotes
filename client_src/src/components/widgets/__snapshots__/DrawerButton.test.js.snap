// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button with material-ui Zoom and Fab components 1`] = `
<div
  className="some-css-class"
>
  <WithTheme(Zoom)
    in={true}
    unmountOnExit={true}
  >
    <WithStyles(Fab)
      aria-label="notes list drawer"
      color="primary"
      onClick={[MockFunction]}
      size="small"
    >
      <pure(MenuIcon) />
    </WithStyles(Fab)>
  </WithTheme(Zoom)>
</div>
`;
